### PR TITLE
Config Reader and TA changes

### PR DESCRIPTION
**Pull request by rashmichandrashekar**

### CODE REVIEW

File: .pipelines/azure-pipeline-build.yml [hunk 1]
```diff
@@ -667,7 +827,7 @@ jobs:
       for i in 1 2 3 4 5 6 7 8 9 10
           do
             sleep 30
-            echo $(MCR_REGISTRY)$(MCR_REPOSITORY):$(IMAGE_TAG_WINDOWS)
+            # echo $(MCR_REGISTRY)$(MCR_REPOSITORY):$(IMAGE_TAG_WINDOWS)
```
Comment (rashmichandrashekar) [inline .pipelines/azure-pipeline-build.yml:830]: remove comment

File: otelcollector/build/linux/configuration-reader/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+COPY --from=builder /usr/bin/inotifywait /usr/bin/inotifywait
+COPY --from=builder /usr/bin/bash /usr/bin/bash
+COPY --from=builder /usr/sbin/busybox /usr/sbin/busybox
+COPY --from=builder /usr/sbin/crond /usr/sbin/crond
+COPY --from=builder /usr/bin/vim /usr/bin/vim
+COPY --from=builder /usr/share/vim /usr/share/vim
+COPY --from=builder /usr/sbin/logrotate /usr/sbin/logrotate
+COPY --from=builder /usr/bin/gzip /usr/bin/
+COPY --from=builder /usr/bin/curl /usr/bin/
+COPY --from=builder /usr/bin/update-ca-trust /usr/bin
```
Comment (rashmichandrashekar) [inline otelcollector/build/linux/configuration-reader/Dockerfile:81]: remove update-ca-cert

**Review by gracewehner — APPROVED** [review]
Thanks Rashmi!! The main.sh scripts refactor looks great

**Comment by rashmichandrashekar** [comment]
> Thanks Rashmi!! The main.sh scripts refactor looks great

No worries! I still haven't tested those changes out. Looks like some scripts are not getting copied.

**Comment by rashmichandrashekar** [comment]
> > Thanks Rashmi!! The main.sh scripts refactor looks great
> 
> No worries! I still haven't tested those changes out. Looks like some scripts are not getting copied.

This is tested out now. Merging the changes.

**Pull request merged**